### ISSUE TITLE
Investigate alternative multipart parsers

### ISSUE THREAD

**Issue by gurgunday** [fastify/fastify-multipart#541]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

For instance [remix parser](https://github.com/mjackson/remix-the-web/tree/main/packages/multipart-parser) seems to be quite faster than `@fastify/busboy` and is async generator based

In theory we should be able to use it, but we should test the speed claims in the context of Fastify

Related to https://github.com/fastify/fastify-multipart/issues/464

**Comment by climba03003** [comment]
From the benchmark script, it seems to buffer chunk into memory by default.
https://github.com/mjackson/multipart-parser/blob/main/bench/parsers/multipart-parser.ts#L22-L24

**Comment by gurgunday** [comment]
Yeah I did some testing and it's really hard to beat fastify/busboy in practical cases, I think it should stay for now

**Comment by gurgunday** [comment]
I'll open this again, some of the concerns like not having a size limit were addressed and their benchmark still shows significant improvements with simpler code.

Even if it's not the case, I'd like to use this issue as a temporary placeholder to investigate alternatives.
